pyuvdata/ms.py

```diff
@@ -0,0 +1,242 @@
+        self.Ntimes=int(len(times_unique))
+        data_array=tb.getcol(data_column)
+        self.Nblts=int(data_array.shape[0])
+        flag_array=tb.getcol('FLAG')
+        #CASA stores data in complex array with dimension NbltsxNfreqsxNpols
+        if(len(data_array.shape)==3):
+            data_array=np.expand_dims(data_array,axis=1)
+            flag_array=np.expand_dims(flag_array,axis=1)
+        self.data_array=data_array
+        self.flag_array=flag_array
+        self.Npols=int(data_array.shape[-1])
+        self.uvw_array=tb.getcol('UVW')
+        self.ant_1_array=tb.getcol('ANTENNA1').astype(np.int32)
+        self.ant_2_array=tb.getcol('ANTENNA2').astype(np.int32)
+        self.Nants_data=len(np.unique(np.concatenate((np.unique(self.ant_1_array),np.unique(self.ant_2_array)))))
+        self.baseline_array=self.antnums_to_baseline(self.ant_1_array,self.ant_2_array)
+        self.Nbls=len(np.unique(self.baseline_array))
+        #Get times. MS from cotter are modified Julian dates in seconds (thanks to Danny Jacobs for figuring out the proper conversion)
+        self.time_array=time.Time(tb.getcol('TIME')/(3600.*24.),format='mjd').jd
+        #Polarization array
+        tbPol=tables.table(filepath+'/POLARIZATION')
+        polList=tbPol.getcol('CORR_TYPE')[0]#list of lists, probably with each list corresponding to SPW. 
+        self.polarization_array=np.zeros(len(polList),dtype=np.int32)
+        for polnum in range(len(polList)):
+            self.polarization_array[polnum]=int(polDict[polList[polnum]])
+        tbPol.close()
+        #Integration time
+        #use first interval and assume rest are constant (though measurement set has all integration times for each Nblt )
+        #self.integration_time=tb.getcol('INTERVAL')[0]
+        #for some reason, interval ends up larger than the difference between times...
+        self.integration_time=float(times_unique[1]-times_unique[0])*3600.*24.
+        #open table with antenna location information
+        tbAnt=tables.table(filepath+'/ANTENNA')
+        tbObs=tables.table(filepath+'/OBSERVATION')
+        self.telescope_name=tbObs.getcol('TELESCOPE_NAME')[0]
+        self.instrument=tbObs.getcol('TELESCOPE_NAME')[0]
+        tbObs.close()
+        #Use Telescopes.py dictionary to set array position
+        self.antenna_positions=tbAnt.getcol('POSITION')
+        xyz_telescope_frame = tbAnt.getcolkeyword('POSITION','MEASINFO')['Ref']
+        antFlags=np.empty(len(self.antenna_positions),dtype=bool)
+        antFlags[:]=False
+        for antnum in range(len(antFlags)):
+            antFlags[antnum]=np.all(self.antenna_positions[antnum,:]==0)
+        try:
+            self.set_telescope_params()
+        except ValueError:#If telescope location is not know, set location manually
+            if(xyz_telescope_frame=='ITRF'):
+                self.telescope_location=np.array(np.mean(self.antenna_positions[np.invert(antFlags),:],axis=0))
```

Comment:
This assumes the positions are in Earth Centered (ECEF) coordinates. If they are just ENU coordinates it won't work.